elixir/apps/web/lib/web/components/form_components.ex

```diff
@@ -404,16 +401,43 @@ defmodule Web.FormComponents do
 
   def edit_button(assigns) do
     ~H"""
-    <.link
-      navigate={@navigate}
-      class="flex items-center justify-center text-white bg-primary-700 hover:bg-primary-800 focus:ring-4 focus:ring-primary-300 font-medium rounded-lg text-sm px-4 py-2 dark:bg-primary-600 dark:hover:bg-primary-700 focus:outline-none dark:focus:ring-primary-800"
-    >
-      <.icon name="hero-pencil-solid" class="h-3.5 w-3.5 mr-2" />
+    <.button style="primary" navigate={@navigate} icon="hero-pencil-solid">
       <%= render_slot(@inner_block) %>
-    </.link>
+    </.button>
     """
   end
 
+  defp button_style do
+    [
+      "flex items-center justify-center",
+      "px-4 py-2 rounded-lg",
+      "font-medium text-sm",
+      "focus:ring-4 focus:outline-none",
+      "phx-submit-loading:opacity-75"
+    ]
+  end
+
+  defp button_style("danger") do
+    button_style() ++
+      [
+        "text-red-600",
+        "border border-red-600",
+        "hover:text-white hover:bg-red-600 focus:ring-red-300",
+        "dark:border-red-500 dark:text-red-500 dark:hover:text-white dark:hover:bg-red-600 dark:focus:ring-red-900"
+      ]
+  end
+
+  defp button_style(_style) do
+    button_style() ++
+      [
+        "text-white",
+        "bg-primary-500",
+        "hover:bg-primary-600",
+        "focus:ring-primary-300",
+        "dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
+      ]
+  end
```

Comment:
This should make updating the classes much less conflict-prone, thanks

Reply:
I wish I could take credit for these functions, but @AndrewDryga is the one that created these earlier, I just moved them to be closer to the button functions.  My contribution to this part of it was more around refactoring the `<.button>` component so it could be the basis for all of the more specific buttons.

I do agree though that this should make it easier to keep styles/themes more consistent and easier to change if/when needed.